src/Databases/DataLake/GlueCatalog.h

```diff
@@ -60,10 +62,18 @@ class GlueCatalog final : public ICatalog, private DB::WithContext
     const LoggerPtr log;
     Aws::Auth::AWSCredentials credentials;
     std::string region;
+    DB::DatabaseDataLakeSettings settings;
+    DB::ASTPtr table_engine_definition;
 
     DataLake::ICatalog::Namespaces getDatabases(const std::string & prefix, size_t limit = 0) const;
     DB::Names getTablesForDatabase(const std::string & db_name, size_t limit = 0) const;
     void setCredentials(TableMetadata & metadata) const;
+
+    /// The Glue catalog does not store detailed information about the types of timestamp columns, such as whether the column is timestamp or timestamptz.
+    /// This method allows to clarify the actual type of the timestamp column.
+    bool classifyTimestampTZ(const String & column_name, const TableMetadata & table_metadata) const;
+
+    mutable Poco::LRUCache<String, Poco::JSON::Object::Ptr> metadata_objects;
```

Comment:
We have our own LRUCache https://github.com/ClickHouse/clickhouse/blob/master/src/Common/LRUCachePolicy.h